src/hotspot/share/cds/filemap.cpp

```diff
@@ -1337,9 +1384,17 @@ bool FileMapInfo::init_from_file(int fd) {
   }
   GenericCDSFileMapHeader* gen_header = file_helper.get_generic_file_header();
 
+  const char* file_type = CDSConfig::type_of_archive_being_loaded();
   if (_is_static) {
-    if (gen_header->_magic != CDS_ARCHIVE_MAGIC) {
-      log_warning(cds)("Not a base shared archive: %s", _full_path);
+    if ((gen_header->_magic == CDS_ARCHIVE_MAGIC) ||
```

Comment:
Probably don't need the extra set of parentheses.

Reply:
Since the next line has a more complex condition, I think putting both this line and the next line in parenthesis makes the code easier to read.